drivers/misc/a4988/a4988.c

```diff
@@ -0,0 +1,247 @@
+/*
+ * Copyright (c) 2023 Kindhome Sp. z o.o <kindhome.io>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT allegromicro_a4988
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/misc/a4988/a4988.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/drivers/gpio.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(a4988);
+
+struct a4988_ms_pins {
+	int ms1;
+	int ms2;
+	int ms3;
+};
+
+struct a4988_dev_config {
+	struct {
+		const struct gpio_dt_spec ms1;
+		const struct gpio_dt_spec ms2;
+		const struct gpio_dt_spec ms3;
+		const struct gpio_dt_spec direction;
+		const struct gpio_dt_spec step;
+		const struct gpio_dt_spec sleep;
+		const struct gpio_dt_spec enable;
+		const struct gpio_dt_spec reset;
+	} gpio;
+};
+
+static int a4988_microstep_to_pins(enum a4988_microstep microstep, struct a4988_ms_pins *ms_pins)
+{
+	switch (microstep) {
+	case a4988_full_step:
+		*ms_pins = (struct a4988_ms_pins){0, 0, 0};
+		return 0;
+	case a4988_half_step:
+		*ms_pins = (struct a4988_ms_pins){1, 0, 0};
+		return 0;
+	case a4988_quarter_step:
+		*ms_pins = (struct a4988_ms_pins){0, 1, 0};
+		return 0;
+	case a4988_eigth_step:
+		*ms_pins = (struct a4988_ms_pins){1, 1, 0};
+		return 0;
+	case a4988_sixteenth_step:
+		*ms_pins = (struct a4988_ms_pins){1, 1, 1};
+		return 0;
+	default:
+		return -EINVAL;
+	}
+}
+
+int a4988_step(const struct device *dev, enum a4988_microstep microstep, bool clockwise)
+{
+	int err;
+	struct a4988_ms_pins ms_pins;
+	const struct a4988_dev_config *config = dev->config;
+
+	err = a4988_microstep_to_pins(microstep, &ms_pins);
+	if (err < 0) {
+		return 0;
+	}
+
+	err = gpio_pin_set_dt(&config->gpio.ms1, ms_pins.ms1);
+	if (err < 0) {
+		return 0;
+	}
+	err = gpio_pin_set_dt(&config->gpio.ms2, ms_pins.ms2);
+	if (err < 0) {
+		return 0;
+	}
+	err = gpio_pin_set_dt(&config->gpio.ms3, ms_pins.ms3);
+	if (err < 0) {
+		return 0;
+	}
+
+	err = gpio_pin_set_dt(&config->gpio.direction, clockwise);
+	if (err < 0) {
+		return 0;
+	}
+
+	k_sleep(K_NSEC(200));
+
+	err = gpio_pin_set_dt(&config->gpio.step, 1);
+	if (err < 0) {
+		return 0;
+	}
+
+	k_sleep(K_USEC(1));
+
+	err = gpio_pin_set_dt(&config->gpio.step, 0);
+	if (err < 0) {
+		return 0;
+	}
+
+	k_sleep(K_USEC(1));
+
+	return 0;
+}
+
+int a4988_sleep(const struct device *dev, bool sleep)
+{
+	const struct a4988_dev_config *config = dev->config;
+	int err;
+
+	if (sleep) {
+		return gpio_pin_set_dt(&config->gpio.sleep, 0);
+	}
+
+	err = gpio_pin_set_dt(&config->gpio.sleep, 1);
+	if (err < 0) {
+		return err;
+	}
+
+	k_sleep(K_MSEC(1));
+
+	return 0;
+}
+
+int a4988_reset(const struct device *dev, bool reset)
+{
+	const struct a4988_dev_config *config = dev->config;
+	int err;
+
+	err = gpio_pin_set_dt(&config->gpio.reset, !reset);
```

Comment:
```suggestion
	err = gpio_pin_set_dt(&config->gpio.reset, reset);
```

There is a gpio flag to indicate a pin is used active low that can be used in an overlay or board dts file. Similar for sleep and enable pins. You should add the active low requirement to the binding description for each gpio pin.